R/utils.R

```diff
@@ -947,15 +949,21 @@ columnToRowData <- function(data) {
 #' shinyApp(ui, server)
 #' }
 #' @export
-customStop <- function(error, errorClass = character(0)) {
+safeError <- function(error, errorClass = character(0)) {
   if (inherits(error, "error")) {
-    class(error) <- c("shiny.custom.error", errorClass, class(error))
-    stop(error)
+    cond <- structure(
+      list(message = error$message),
+      class = c("shiny.custom.error", errorClass, class(error))
+    )
   } else if (inherits(error, "character")) {
-    stopWithCondition(c("shiny.custom.error", errorClass), error)
+    cond <- structure(
+      list(message = error),
+      class = c("shiny.custom.error", errorClass, "error", "condition")
+    )
```

Comment:
Ok, that makes sense. I've changed to that, with a small difference: I don't think this check is necessary: `if (!("shiny.custom.error" %in% class(error))`. `error` will never have the class `shiny.custom.error` a priori -- we'll always have to add it (right?) Also, I added the user-provided `errorClass` to `class(error)`.